### PR TITLE
Feature/fbfw 114 support move graphics

### PR DESCRIPTION
**Pull request by bryancodazen**
### Summary
This PR adds the logic to draw arrows based on existing `ordersMeta` data. The logic is able to distinguish when the supporting unit and supported unit's orders don't match and in that case will draw an implied arrow (gray). If the orders do match it will draw a support arrow based on the mocks.  

Additionally, there are a few other updates not directly related to support move but which improve the codebase overall. 
1. Drawing specific orders has been removed from WDMapController.tsx
2. There is a new global state object called `maps` which holds "lookup" information so that we can make the connection between a unit, territory, etc much easier. 
For example, if you wanted to know what territory a unit is in and you have the `unitID` you can do `const terrID = maps.unitToTerritory[unitID]` and this will give you the territoryID that unit is in. 

### Video
https://drive.google.com/file/d/12EO3WyQHG6OvIDibQApHkZ2gzmwIQ5qR/view?usp=sharing

In the video I change orders around so that you can see how it works with different orders and how depending on the order information it draws the implied arrow.

### CODE REVIEW

File: beta-src/src/state/interfaces/GameStateMap.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import GameStateMap from "../../types/state/GameStateMap";
```
Comment (mariomelchor) [inline beta-src/src/state/interfaces/GameStateMap.ts:1]: Nit: Rename File to GameStateMaps.ts (plural to match interface name)

**Review by justinyoocodazen — APPROVED** [review]

File: beta-src/src/utils/map/drawMoveOrders.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+export default function drawMoveOrders(data, ordersMeta: OrdersMeta): void {
+  const { currentOrders, territories, units } = data;
+  const ordersMetaEntries = Object.entries(ordersMeta);
+  if (ordersMetaEntries.length && units && territories) {
+    ordersMetaEntries
+      .filter(([, { update }]) => update?.type === "Move")
+      .forEach(([orderID, { update }]) => {
+        const originalOrder = currentOrders.find(({ id }) => {
+          return id === orderID;
+        });
```
Comment (mariomelchor) [inline beta-src/src/utils/map/drawMoveOrders.ts:16]: implicit return
` const originalOrder = currentOrders.find(({ id }) =>  id === orderID);`

File: beta-src/src/utils/map/drawSupportMoveOrders.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+): void {
+  const { currentOrders, territories, units } = data;
+  const ordersMetaEntries = Object.entries(ordersMeta);
+  if (ordersMetaEntries.length && units && territories) {
+    ordersMetaEntries
+      .filter(([, { update }]) => update?.type === "Support move")
+      .forEach(([orderID, { update }]) => {
+        const originalOrder = currentOrders?.find(({ id }) => {
+          return id === orderID;
+        });
```
Comment (mariomelchor) [inline beta-src/src/utils/map/drawSupportMoveOrders.ts:23]: implicit return here

**Review by mariomelchor — CHANGES_REQUESTED** [review]

File: beta-src/src/state/interfaces/GameStateMap.ts (hunk 1 above)
Comment (bryancodazen) [reply beta-src/src/state/interfaces/GameStateMap.ts:1]: `GameStateMaps` and `GameStateMap` are different. 

`GameStateMaps` is composed of multiple `GameStateMaps`

Comment (mariomelchor) [reply beta-src/src/state/interfaces/GameStateMap.ts:1]: Sorry let me clarify. I meant to rename this file from `GameStateMap.ts` to `GameStateMaps.ts`

**Review by mariomelchor — CHANGES_REQUESTED** [review]

File: beta-src/src/state/interfaces/GameStateMap.ts (hunk 1 above)
Comment (bryancodazen) [reply beta-src/src/state/interfaces/GameStateMap.ts:1]: Ahh I  see the issue, there is a discrepancy in the file names. will update

**Review by mariomelchor — APPROVED** [review]
LGTM

**Pull request merged**